script.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+let total = 0;
+let highest = 0;
+let stuAvg= 0;  
+var names= [];
+var grade= [];
+var pos1 = [];
+var arr = [];
+
+for (var i = 0; i < 10; i++) {             
+    names.push(prompt('Enter The Ten Student Names | Student: ' + (i + 1))); 
+    document.getElementById("stuNames").innerHTML=("Student's Names:" +names.join(','));
+          
+    grade.push(Number(prompt('Enter The Ten Student Grades | Grade: ' + (i + 1))));
+    total += grade[i] 
+      
+    stuAvg = total / grade.length;
+    highest = Math.max(...grade);
+    pos1 = grade.indexOf(highest);
+    document.getElementById("stuGrades").innerHTML=("Student's Grades:" +grade.join(','));
+}
+
+
+function results() {
+    document.getElementById("result").innerHTML=("Highest Grade:"+highest+"The Highest Grade Belongs to:"+names[pos1]+" The Average Grade is "+stuAvg+"");
+}  
+
+function passingStu() {
+
+    grade.filter(passing); 
+    function passing(grade) {
+            return grade[i] >= 60;      
+    }
+    document.getElementById("passingStu").innerHTML=("The Students who are passing are: "+grade.filter(passing)+""+names[i]+"");

```

Comment:
Where is this '[i]' coming from? Is this function being defined in the for loop? Can't tell by the indentation.

Reply:
the [i] on line 33? it was just an attempt to see if I call the grade indices there, would they pass into names[i]

Reply:
No, the scope of the i variable was limited to the for loop way before. You could start a new for loop to get access to a new index variable but at this point unless you passed `i` as an argument to the passing function, there wouldn't be any `i` variable in the scope of this function.

Reply:
You'll have to make that new passingStudents array, but then afterwards you'll have to make another for loop anyway to check which students you'll be passing to the passingStudents array.